src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,347 @@
+            /// <returns>A task representing the asynchronous flush operation.</returns>
+            public override async Task FlushAsync(CancellationToken cancellationToken)
+            {
+                await this.jsonWriter.FlushAsync().ConfigureAwait(false);
+            }
+
+            /// <summary>
+            /// Disposes the object.
+            /// </summary>
+            /// <param name="disposing">True if called from Dispose; false if called form the finalizer.</param>
+            protected override void Dispose(bool disposing)
+            {
+                if (this.numBytesNotWrittenFromPreviousChunk > 0)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(0, this.numBytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk, isFinalBlock: true, out numBytesNotWrittenFromPreviousChunk);
+                    this.numBytesNotWrittenFromPreviousChunk = 0;
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(this.buffer);
+                }
+
+                this.Flush();
+                base.Dispose(disposing);
+            }
+
+            /// <summary>
+            /// Asynchronously disposes of the current object and performs cleanup operations.
+            /// </summary>
+            /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+            public override async ValueTask DisposeAsync()
+            {
+                if (this.numBytesNotWrittenFromPreviousChunk > 0)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(0, this.numBytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk.Span, isFinalBlock: true, out numBytesNotWrittenFromPreviousChunk);
+                    this.numBytesNotWrittenFromPreviousChunk = 0;
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(this.buffer);
+                }
+
+                await this.jsonWriter.FlushAsync().ConfigureAwait(false);
+            }
+
+            /// <summary>
+            /// Reads a sequence of bytes from the current stream and advances the position within the stream by the numbers of bytes read. This operation is not supported by this stream.
+            /// </summary>
+            public override int Read(byte[] buffer, int offset, int count)

```

Comment:
nit: You could use the shorter form for this method and others where operation is not supported...
```csharp
public override int Read(byte[] buffer, int offset, int count) => throw new NotSupportedException();
```

Reply:
I think this is the method that gets included by default when this class derives from `Stream`